### PR TITLE
Add NotNullWhen(true) on Directory/File.Exists

### LINKED ISSUE #47573: Directory.Exists and File.Exists annotated parameter with [NotNullWhen(true)]
## Background and Motivation

Found when annotating roslyn. These methods should never return `true` for null.

## Proposed API

```diff
namespace System.IO
{
    public static class Directory {
-        public static bool Exists(string? path);
+        public static bool Exists([NotNullWhen(true)] string? path);
    }
    public static class File {
-        public static bool Exists(string? path);
+        public static bool Exists([NotNullWhen(true)] string? path);
    }
}
```

## Usage Examples

``` C#
string? path = ...
if (Directory.Exists(path))
{
    var subDirectories = Directory.GetDirectories(path); // no warning
}
```

## Alternative Designs

None.

## Risks

I couldn't imagine how they can return `true` for null.

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Closes #47573 .

Note: there were already annotations on **ref of Directory** and **impl on File**. Doesn't ApiCompact check this?

### CODE REVIEW

File: src/libraries/System.IO.FileSystem/ref/System.IO.FileSystem.cs [hunk 1]
```diff
@@ -126,7 +126,7 @@ public static partial class File
         public static void Delete(string path) { }
         [System.Runtime.Versioning.SupportedOSPlatformAttribute("windows")]
         public static void Encrypt(string path) { }
-        public static bool Exists(string? path) { throw null; }
+        public static bool Exists([System.Diagnostics.CodeAnalysis.NotNullWhen(true)] string? path) { throw null; }
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem/ref/System.IO.FileSystem.cs:129]:
```suggestion
        public static bool Exists([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] string? path) { throw null; }
```

File: src/libraries/System.IO.FileSystem/src/System/IO/Directory.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using System.IO;
 using System.Collections.Generic;
 using System.Linq;
+using System.Diagnostics.CodeAnalysis;
```
Comment (stephentoub) [inline src/libraries/System.IO.FileSystem/src/System/IO/Directory.cs:8]: System.Collections.Generic was already out of order, but can you move this up to be alphabetical?

**Review by stephentoub — APPROVED** [review]

**Comment by stephentoub** [comment]
> there were already annotations on ref of Directory and impl on File. Doesn't ApiCompact check this?

@ericstj, @safern?

**Comment by safern** [comment]
APICompat doesn't have a way to check nullable annotations yet, however we do plan on fixing this for the 6.0.0 release.

**Pull request merged**